router-tests/utils.go

```diff
@@ -35,7 +36,7 @@ func configureAuth(t *testing.T) ([]authentication.Authenticator, *jwks.Server)
 	authServer, err := jwks.NewServer(t)
 	require.NoError(t, err)
 	t.Cleanup(authServer.Close)
-	tokenDecoder, _ := authentication.NewJwksTokenDecoder(zap.NewNop(), authServer.JWKSURL(), time.Second*5)
+	tokenDecoder, _ := authentication.NewJwksTokenDecoder(context.Background(), zap.NewNop(), authServer.JWKSURL(), time.Second*5)
```

Comment:
Would it be better to create a helper func that takes *testing.T and returns a context that gets cancelled when t.Cleanup is called? I'm not sure about the token decoder implementation, so it might or might not have any meaningful effect.

Reply:
https://github.com/wundergraph/cosmo/pull/1488/files#diff-f705492dfd8e245ece0ac151b18e19be3818b88ebf3857da8b9fcd07f0ca3890R54
I think it could be beneficial

Reply:
Good point, I'm gonna move it into the test func and use the global environment context.

Reply:
Done.